### PR TITLE
fix: merge conflict dialog added when overwriting files on move-file action

### LINKED ISSUE #43489: [Bug]: When moving a folder, no merge dialog appears if the folder names are the same.
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I move a folder with name d1 in a new destination where a  folder with the same name (d1) exist, there came non merge dialog. If the two folders have different contents, these are overwritten by the contents to be moved. 

### Steps to reproduce

1. Open files App
2. select folder d1 
3. run action copy or move
4. select destination folder
5.  run move action

### Expected behavior

Wenn die Inhalte der gleich namigen Ordner  die Aktion verschieben ausgeführt wird sollte der zusammenführen Dialog erscheinen. Mit dem man ein überschreiben oder zusammenführen der Ordner Auswählen kann. 

### Installation method

Community Docker image

### Nextcloud Server version

28

### Operating system

Other

### PHP engine version

None

### Web server

Other

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "htaccess.RewriteBase": "\/",
        "filelocking.enabled": true,
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "...."
        ],
        "overwrite.cli.url": "https:\/\/cloud.com",
        "overwriteprotocol": "https",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "28.0.2.5",
        "dbtableprefix": "oc_",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "theme": "",
        "loglevel": 2,
        "log_type": "file",
        "syslog_tag": "",
        "logdateformat": "F d, Y H:i:s",
        "logfile": "",
        "maintenance": false,
        "default_phone_region": "DE",
        "onlyoffice": {
            "verify_peer_off": true
        },
        "app_install_overwrite": [
            "initialcheck",
            "limit_login_to_ip",
            "radio",
            "tasks",
            "occweb",
            "files_reader",
            "files_ebookreader",
            "polls",
            "contacts",
            "drawio",
            "checksum",
            "rainloop",
            "timetracker",
            "passman",
            "calendar",
            "cookbook",
            "dashboardcharts",
            "apporder",
            "bookmarks",
            "passwords",
            "carnet",
            "electronicsignatures",
            "ocdownloader",
            "files_bpm",
            "gestion",
            "audioplayer_editor",
            "news"
        ],
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\HEIC",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\Movie",
            "OC\\Preview\\PDF",
            "OC\\Preview\\Illustrator"
        ],
        "app.mail.imap.timeout": 20,
        "app.mail.smtp.timeout": 2,
        "app.mail.verify-tls-peer": "false",
        "has_rebuilt_cache": true,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "updater.release.channel": "stable",
        "data-fingerprint": "98",
        "auth.bruteforce.protection.enabled": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by emoral435**
* Resolves #43489

## Summary
* When user decides to move file to new destination, if a file with that same name already exists, currently, our implementation overrides the old file and injects the new file

## Resolution
* In order to resolve this, use @nextcloud/upload to check if there is a conflict, and if there is, shows a conflict dialog to resolve how the user wishes to proceed with the conflicting dialogs
* If the conflict dialog returns two empty arrays, selected and renamed, that means the user chose to keep the file already in the directory path, thereby choosing to delete the current selected file they want to move

## Screenshots / GIF
firefox_1XwZZSoVxP

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/actions/moveOrCopyAction.ts [hunk 1]
```diff
@@ -133,6 +135,27 @@ export const handleCopyMoveNodeTo = async (node: Node, destination: Folder, meth
...
+						// if the user selected to keep the old file, and did not select the new file
+						// that means they opted to delete the current node
+						if (!selected.length && !renamed.length) {
+							await client.deleteFile(currentPath)
+							emit('files:node:deleted', node)
+							return
+						}
+					} catch (error) {
+						// User cancelled
+						showError(t('files','Upload cancelled'))
```
Comment (Pytal) [inline apps/files/src/actions/moveOrCopyAction.ts:153]:
```suggestion
						showError(t('files','Move cancelled'))
```

**Comment by Pytal** [comment]
Not related to this change but the conflict dialog close button is focusable with the keyboard but unclickable
image
I believe this is an issue with the z-index

File: apps/files/src/actions/moveOrCopyAction.ts (hunk 1 above)
Comment (emoral435) [reply apps/files/src/actions/moveOrCopyAction.ts:153]: Nice catch! Changing to `Move`.

**Review by skjnldsv — APPROVED** [review]
Logic makes sense!

**Review by susnux — APPROVED** [review]

**Pull request merged**